app/controllers/companies_controller.rb

```diff
@@ -1,53 +1,19 @@
-class CompaniesController < ApplicationController
-  before_action :set_company, only: [:show, :update, :destroy]
+class CompaniesController < CrudController
+  self.permitted_attrs = %i[name web email phone
+                            partnermanager contact_person email_contact_person
+                            phone_contact_person crm level my_company]
+
+  self.nested_models = %i[locations employee_quantities people]
 
-  # GET /companies
   def index
     @companies = Company.all
-
     render json: @companies
   end
-
-  # GET /companies/1
-  def show
-    render json: @company
-  end
-
-  # POST /companies
-  def create
-    @company = Company.new(company_params)
-
-    if @company.save
-      render json: @company, status: :created, location: @company
-    else
-      render json: @company.errors, status: :unprocessable_entity
-    end
-  end
-
-  # PATCH/PUT /companies/1
-  def update
-    if @company.update(company_params)
-      render json: @company
-    else
-      render json: @company.errors, status: :unprocessable_entity
-    end
-  end
-
-  # DELETE /companies/1
+  
   def destroy
-    @company.destroy
-  end
-
-  private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_company
-      @company = Company.find(params[:id])
-    end
-
-    # Only allow a trusted parameter "white list" through
-    def company_params
-      params.require(:company).permit(:name, :web, :email, :phone, :partnermanager, 
-                                      :contact_person, :email_contact_person,
-                                      :phone_contact_person, :crm, :level, :picture, :my_company)
+    company = Company.find(params[:id])
+    unless company.my_company
+      company.destroy
```

Comment:
move that check into the model class.
raise an exception if someone tries to delete his own company.
use .destroy! (check difference between #destroy and #destroy!)
you probably then can use standard crud destroy